src/main/java/uk/gov/hmcts/reform/pip/account/management/database/AuditRepository.java

```diff
@@ -5,14 +5,22 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.transaction.annotation.Transactional;
 import uk.gov.hmcts.reform.pip.account.management.model.AuditLog;
+import uk.gov.hmcts.reform.pip.model.enums.AuditAction;
 
 import java.time.LocalDateTime;
+import java.util.List;
 import java.util.UUID;
 
 public interface AuditRepository extends JpaRepository<AuditLog, UUID> {
 
     @Transactional
     void deleteAllByTimestampBefore(LocalDateTime timestamp);
 
-    Page<AuditLog> findAllByOrderByTimestampDesc(Pageable pageable);
+    Page<AuditLog> findAllByUserEmailLikeIgnoreCaseAndUserIdLikeAndActionInAndTimestampBetweenOrderByTimestampDesc(
```

Comment:
I am not sure it is better but use the default JPA method provided by spring boot here, or use our owned constructed SQL query. The sticking point here is the timestamp, where we have to pass in an arbitrary start and end time from the front end. If we are using our owned query, we can pass in null and have the query check for null timestamp or a date timestamp by stripping off the time bit from the data. Happy to discuss :)

Reply:
As discussed this morning in Dev Standup, it's fixed.